lib/schema-controller.js

```diff
@@ -79,6 +79,14 @@ class SchemaController {
     return this.serializerCompiler || (this.parent && this.parent.getSerializerCompiler())
   }
 
+  getSerializerBuilder () {
+    return this.compilersFactory.buildSerializer || (this.parent && this.parent.getSerializerBuilder())
```

Comment:
I must thing why this is needed

This line 
https://github.com/fastify/fastify/blob/main/lib/pluginOverride.js#L51

should do the trick already

Reply:
Hmm, true! But the issue is on this line: https://github.com/fastify/fastify/blob/6a0403319148163fe6f826aba16e33e46fdb88d4/fastify.js#L639

For instance, if a user wants to override the SchemaControler, as the new SchemaController opts are at the rightmost, this will take precedence over the current parent SchemaController opts.
The only way to avoid such overwritten is to make a deep copy, but not sure if really worth it. What do you think?